### PR TITLE
feat: configure MCP servers in OpenCode agent preWorkspaceStart

### LINKED ISSUE #2232: Update opencode extension to handle MCP servers
### Is your feature request related to a problem? Please describe

The agent preWorkspaceStart should configure MCP

### Describe the solution you'd like

See https://opencode.ai/docs/mcp-servers/

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
## Summary
- Translate workspace MCP servers (remote) and commands (local) into OpenCode's native `mcp` config format during the `preWorkspaceStart` hook
- Add `McpEntrySchema` with typed fields (`type`, `url`, `command`, `enabled`, `environment`) to validate the OpenCode MCP config
- Merge workspace MCP entries with any existing MCP config in `opencode.json`

Fixes #2232

## Test plan
- [x] Unit tests for remote MCP servers (with and without headers)
- [x] Unit tests for local MCP commands (with and without args/env)
- [x] Unit tests for mixed remote + local servers
- [x] Unit tests for merging with existing MCP config
- [x] Unit tests for edge cases (empty headers/env omitted, no MCP preserves existing)
- [x] All 26 tests pass
- [x] TypeScript compiles cleanly
- [x] ESLint reports no issues

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: extensions/opencode/src/extension.ts [hunk 1]
```diff
@@ -106,6 +117,33 @@ export async function activate(extensionContext: ExtensionContext): Promise<void
...
+
+      if (mcpServers?.length || mcpCommands?.length) {
+        const mcp: Record<string, McpEntry> = { ...config.mcp };
+
+        for (const server of mcpServers ?? []) {
+          mcp[server.name] = {
+            type: 'remote',
+            url: server.url,
+            enabled: true,
+            ...(server.headers && Object.keys(server.headers).length > 0 ? { environment: server.headers } : {}),
```
Comment (MarsKubeX) [inline extensions/opencode/src/extension.ts]: For type "remote" instead of "environment" the schema uses "headers": https://opencode.ai/docs/mcp-servers/

```suggestion
            ...(server.headers && Object.keys(server.headers).length > 0 ? { headers: server.headers } : {}),
```

File: extensions/opencode/src/extension.ts [hunk 2]
```diff
@@ -34,9 +34,20 @@ const ProviderEntrySchema = z.looseObject({
   models: z.record(z.string(), ModelEntrySchema).default({}),
 });
 
+const McpEntrySchema = z.looseObject({
+  type: z.enum(['remote', 'local']),
+  url: z.string().optional(),
+  command: z.array(z.string()).optional(),
+  enabled: z.boolean(),
+  environment: z.record(z.string(), z.string()).optional(),
+});
```
[the remark this answers is not in the document]
Comment (MarsKubeX) [reply extensions/opencode/src/extension.ts]: Seems valid to me.

**Review by MarsKubeX — CHANGES_REQUESTED** [review]

**Review by MarsKubeX — APPROVED** [review]
LGTM
